core/modules/wmgr/python/dbMgr.py

```diff
@@ -810,7 +944,7 @@ def getIndex(dbName, tblName, chunkId=None):
             if not rows:
                 break
             for row in rows:
-                result.append(list(row))
+                result.append(tuple(row))
 
```

Comment:
It would be fine to have a documented, high-level picture of data-loader architecture. Indeed it becomes quite complex now. Could you do it soon in a new ticket please?